spec/rubocop/cop/rspec/leaky_constant_declaration_spec.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::LeakyConstantDeclaration do
+  subject(:cop) { described_class.new }
+
+  describe 'constant assignment' do
+    it 'flags inside an example group' do
+      expect_offense(<<-RUBY)
+        describe SomeClass do
+          CONSTANT = 'is not allowed'.freeze
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Stub constant instead of declaring explicitly.
+        end
+      RUBY
+    end
+
+    it 'flags inside shared example group' do
+      expect_offense(<<-RUBY)
+        RSpec.shared_examples 'shared example' do
+          CONSTANT = 'is not allowed'.freeze
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Stub constant instead of declaring explicitly.
+        end
+      RUBY
+    end
+
+    it 'flags inside an example' do
+      expect_offense(<<-RUBY)
+        describe SomeClass do
+          specify do
+            CONSTANT = 'is not allowed'.freeze
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Stub constant instead of declaring explicitly.
+          end
+        end
+      RUBY
+    end
+
+    it 'ignores outside of example/shared group' do
+      fair_code = <<-RUBY
+        factory :some_class do
```

Comment:
Does it make sense to submit this cop to core RuboCop? Are there some legit cases for constant declaration in a block?

```
c = Class.new do
  A = 1
  def a
    A
  end
end

# Pretty much as expected
c.new.a # => 1

# Totally unexpected
A # => 1
```

Reply:
Good question. You could try it out on some open source projects, see if you find any violations there.

Reply:
It turns outs the vast majority of the offences are in specs.